lib/stat_tracker.rb

```diff
@@ -21,6 +21,35 @@ def initialize(files)
     end
   end
 
+  def team_list
+    @teams.each_with_object(Hash.new) { |team, team_list| team_list[team.team_id] = team.team_name}
+  end
+  
+  def best_offense
+    goals_made = @game_teams.each_with_object(Hash.new(0.0)) { |game, goals_made| goals_made[game.team_id] += game.goals.to_i}
+    games_played_per_team = @game_teams.each_with_object(Hash.new(0.0)) { |game, goals_made| goals_made[game.team_id] += 1}
+    average_goals_per_team = Hash.new(0.0)
+    goals_made.each do |key, value|
```

Comment:
Nice logic! Breaking down this problem into actionable steps is the key to method creation. There's room for refactor using the each_with_object as it allows iteration and object creation at the same time for line 31&32. Pseudocode can be helpful to break down logic into individual pieces.